### DISCUSSION TITLE
user roles on items?

### DISCUSSION THREAD

**Discussion by Mariusthvdb** [elchininet/custom-sidebar#245] [Ideas]
having finished my customizations on the regular menu items, there is 1 thing I can not get my head around:

how can I set a visibility for admin only on my menu items, and not c&p the full config in an exceptions config based on usernames?

would it be feasible to have something like

```
  - item: ui-familie
    match: data-panel
    order: 3
    role: admin
```

or 
```
  - item: ui-welcome
    match: data-panel
    order: 3
    role: user
```

or maybe `admin: true` and adhere to the user roles/groups from the main config settings in /config/users ?

that is what [`custom:state-switch`](https://github.com/thomasloven/lovelace-state-switch) uses

> group
> 
> group can take one of two values - admin or user depending on whether the currently logged in user is in the Administrators group or not.

**Comment by elchininet** [comment]
>how can I set a visibility for admin only on my menu items, and not c&p the full config in an exceptions config based on usernames?

There is no way to set admin/not admin configurations at the moment. But it is not needed to copy and paste the whole configuration. If you set `extend_from_base` as true, it will clone the base config, and optionally you can add more configurations to merge with it.

```yaml
## Admin 1, Admin 2 and Admin 3 will have the same config as in base
exceptions:
  extend_from_base: true
  - user:
    - Admin 1
    - Admin 2
    - Admin 3
```

**Reply by Mariusthvdb** [reply]
Guess  I don’t understand yet. I  want to limit visibity for several dashboards to certain users (admins in this case).

could you please give an item example how to do that ?

**Reply by elchininet** [reply]
I understood totally the oposite, you mentioned "how can I set a visibility for admin only on my menu items" and I thought that you wanted to show things that were hidden for other admins, but what you want is to hide things that are already visible.

Then it seems that you need to reverse your logic. What you put in the main config is what you want for everybody and the exceptions are just that, "exceptions". You should not put in the main config a configuration that you want just for you and start to create several exceptions to hide the things that you don't want to be visible for others (because in that case they will not be exceptions).

If you want to hide things for several users (or for everybody) and just show things for a few, put in the main config the configuration that you want by default, and then create an exception just for you or for the users that you decide.

I have an option in development that would help with that, it is basically an option to hide all the items by default, and in the order array you specify which ones you want to show.

**Reply by Mariusthvdb** [reply]
sorry if I was unclear.... guess I just ask for this `require_admin`, [like HA provides](https://www.home-assistant.io/dashboards/dashboards/#require_admin):
```
    ui-develop:
      mode: yaml
      filename: dashboard/dashboards/ui-develop.yaml
      title: Systeem info
      icon: mdi:home-assistant
      require_admin: true
```

so, show for admins, and Not show for non-admins. (or, users, in the state_switch terminology)
<img width="880" alt="Scherm­afbeelding 2024-12-06 om 12 12 25" src="https://github.com/user-attachments/assets/9d57ba75-d59f-40b6-8ea7-13f951decd7d">

ofc we can set that in HA, but as you said, better not have the 2 systems interfere.

**Reply by elchininet** [reply]
Hi @Mariusthvdb,
Different implementations and therefore different APIs. Custom Sidebar's API is based on exceptions, user access is not set by individual items but by exceptions and implementing different user access will not only lead to code complexity but also to bugs sooner rather than later.  I'll release [this](https://github.com/elchininet/custom-sidebar/pull/247) soon, maybe during the weekend, it would give you more flexibility to set exceptions based on admin roles.

>ofc we can set that in HA, but as you said, better not have the 2 systems interfere.

There is no interference in this case, you can use that if you prefer. The only thing that would interfere with custom sidebar is to hide/reorder items using the core functionality.
